frontend/src/app/auth.service.ts

```diff
@@ -19,12 +19,12 @@ export class AuthService {
   private logoutUrl = `${environment.apiRoot}/logout/`;
   private resetPasswordUrl = `${environment.apiRoot}/password-reset/`;
 
-  private changeUserEvent = new BehaviorSubject<User | null>(null);
+  private changeUserEvent = new ReplaySubject<User>(1);
```

Comment:
Quick question here: Was there a problem with BehaviorSubject or was there any other reason to change it to ReplaySubject? Also, it seems to me that the type returned still can be <User | null > although compiler might have gotten lost here.

Reply:
You're right about the type.

Let me answer about the question about subjects since I asked Julia to try it out. The issue with `BehaviorSubject` here is that it keeps and emits the initial value (`null` in this case) to all the subscribers, even before we actually get the user info from the backend. `ReplaySubject` does not hold initial value and only emits after the first event comes.

So, normally, when entering the page `BehaviorSubject` would emit `null` (which makes the guard redirect to login) and only then emit the user (if there is any) - which is too late.

I think the user subject should not emit anything before we actually know if there is any user.

It might also lead to other unexpected page behaviors, i.e. if we supposedly had an example stream like `isUserOrgMember$ = organization$.combineLatest(user$).map(...blah blah...)`, it might emit two opposite values (which might cause i.e. some blinking in the view), whereas with `ReplaySubject` it would wait for the user info - which is correct IMO.

The issue might be solved in other ways and TBH I'm 100% sure if we don't break that way anything that relies on `user$` (I think we shouldn't), so we need some check here.